metrics/network/remote_network/remote-networking-memory-smem.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This script will measure RSS memory with smem tool while running an iperf3
+# bandwidth network measurement using a remote setup, where host A will run a
+# server container and host B will run a client container.
+
+set -e
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/remote-networking-test-common.sh"
+
+# Set QEMU_PATH unless it's already set
+QEMU_PATH=${QEMU_PATH:-$(get_qemu_path)}
+# Time in which we sample the memory (seconds)
+middle_time=10
+# Time to run the server using iperf3 (seconds)
+server_time=30
+
+# This function describes how to use this script
+function help {
+echo "$(cat << EOF
+Usage: $0 "[options]"
+	Description:
+		This script will measure RSS memory using smem tool while running
+		a network bandwidth measurement with iperf3 with a remote setup, where host A
+		will run a server container and host B will run a client container.
+		In order to run this script, these inputs are needed:
+		- Interface name where swarm will run.
+		- User of the host B.
+		- IP address of the host B
+	Options:
+		-h      Shows help
+		-r      Run RSS memory
+		-i      Interface name to run Swarm (mandatory)
+		-u      User of host B (mandatory)
+		-a      IP address of host B (mandatory)
+
+EOF
+)"
+}
+
+# This function will measure the bandwidth using iperf3
+function remote_network_rss_memory {
+	setup_swarm
+	client_replica_status
+	server_replica_status
+
+	server_ip_address=$(check_server_address)
+	start_server
+
+	client_id=$($DOCKER_EXE ps -q)
+	client_command="mount -t ramfs -o size=20M ramfs /tmp && iperf3 -c $server_ip_address -t $server_time"
+	start_client "$client_id" "$client_command" > /dev/null
+
+	# Time when we are taking our RSS measurement
+	echo >&2 "WARNING: sleeping for $middle_time seconds in order to sample the RSS"
+	sleep ${middle_time}
+
+	process="$QEMU_PATH"
+	memory_command="sudo smem --no-header -c rss"
+	result=$(${memory_command} -P ^${process})
+
+	memory=$(echo "$result" | awk '{ total += $1 } END { print total/NR }')
+	echo "RSS Memory is : $memory Kb"
```

Comment:
Do we not want to use the standardised 'save results routine to get the data into the csv files etc.?

Reply:
@grahamwhaley why do we need to save the results in csv files? This is not going to inject results in a database

Reply:
Hi @GabyCT , as @grahamwhaley say, this script should save the metrics results in CSV files in order to be analyzed/processed (no necessarily for being saved in a database but it could be, is easier to process CSV data then raw data). if you take a look to other scripts they are standardized with that, even we have a tool that can help us with that https://github.com/clearcontainers/tests/tree/master/cmd/checkmetrics

I am not sure if previous/recent scripts include that, if not, could be great a fix for that :)